projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts

```diff
@@ -80,9 +80,12 @@ export class NgxIndexedDBService {
               ? objectStore.add(value, key)
               : objectStore.add(value);
 
-            request.onsuccess = (evt: Event) => {
-              const result = (evt.target as IDBOpenDBRequest).result;
-              resolve((result as unknown) as number);
+            request.onsuccess = async (evt: Event) => {
+              const result: any = (evt.target as IDBOpenDBRequest).result;
+              const getRequest: IDBRequest = objectStore.get(result) as IDBRequest<T>;
```

Comment:
Shouldn't we be using `this.getByKey`?

Reply:
Hi @assuncaocharles,
The getByKey method if called internally to another method returns promise resolved.